extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/cors/CORSFilter.java

```diff
@@ -90,29 +89,36 @@ public void handle(RoutingContext event) {
                 processRequestedHeaders(response, requestedHeaders);
             }
 
-            boolean allowsOrigin = corsConfig.origins.isEmpty() || corsConfig.origins.contains(origin);
+            boolean allowsOrigin = containsOnlyWildcard(corsConfig.origins) || corsConfig.origins.contains(origin);
 
             if (allowsOrigin) {
                 response.headers().set(HttpHeaders.ACCESS_CONTROL_ALLOW_ORIGIN, origin);
             }
 
             response.headers().set(HttpHeaders.ACCESS_CONTROL_ALLOW_CREDENTIALS, "true");
 
-            final String exposedHeaders = corsConfig.exposedHeaders.stream().collect(Collectors.joining(","));
+            if (!corsConfig.exposedHeaders.isEmpty()) {
+                List<String> exposedHeaders = corsConfig.exposedHeaders;
+                if (corsConfig.exposedHeaders.contains("*") && corsConfig.exposedHeaders.size() > 1) {
+                    exposedHeaders.remove("*");
+                }
 
-            if (!exposedHeaders.isEmpty()) {
-                response.headers().set(HttpHeaders.ACCESS_CONTROL_EXPOSE_HEADERS, exposedHeaders);
+                response.headers().set(HttpHeaders.ACCESS_CONTROL_EXPOSE_HEADERS, String.join(",", exposedHeaders));
             }
 
             if (request.method().equals(HttpMethod.OPTIONS)) {
                 if ((requestedHeaders != null || requestedMethods != null) && corsConfig.accessControlMaxAge.isPresent()) {
-                    response.putHeader(HttpHeaders.ACCESS_CONTROL_MAX_AGE,
-                            String.valueOf(corsConfig.accessControlMaxAge.get().getSeconds()));
+                    response.putHeader(HttpHeaders.ACCESS_CONTROL_MAX_AGE, String.valueOf(corsConfig.accessControlMaxAge.get()
+                            .getSeconds()));
                 }
                 response.end();
             } else {
                 event.next();
             }
         }
     }
+
+    private boolean containsOnlyWildcard(List<String> list) {
+        return list.size() == 1 && "*".equals(list.get(0));
```

Comment:
I asked for it to support the empty list and I don't think you have addressed it.

Reply:
done